### DISCUSSION TITLE
Connectivity issue KingKong Mini 2 with Dexcom G6

### DISCUSSION THREAD

**Discussion by dicko72** [NightscoutFoundation/xDrip#2918] [Q&A] [answered]
Hi all,
Been using xdrip for many years and a couple of months back connectivity has become very intermittent.
Readings are missed for long periods of time then recover without intervention.
Data is backfilled.
All power saving stuff like duraspeed etc is turned off and all permissions and background power settings set correctly.
I can get readings when I use the BYODA Dexcom app.
Any tips where to start further troubleshooting?
Running Android 10 on a Cubot KingKong Mini 2.
Thanks

**Comment by Navid200** [comment]
Do you have "Avoid scanning" disabled?

You say you can get readings when you use the Dexcom app.  But, your issue with xDrip is not all the time.  As you describe it is intermittent.  
Have you used the Dexcom app consistently long enough to see if that may also have intermittent connectivity?

**Comment by jamorham** [comment]
I've tested with KingKong mini 2 and once duraspeed was disabled it was rock solid as a collector using the default settings.

**Comment by dicko72** [comment]
It always used to be pretty good. Just not sure why now it is not working.

**Reply by Navid200** [reply]
You can go over the settings to make sure nothing has been accidentally set incorrectly:
https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

**Comment by dicko72** [comment]
Yes these are the exact settings I use. An excellent resource. I can
confirm the settings do match.

**Reply by Navid200** [reply]
When was the last Android update/patch?

**Reply by dicko72** [reply, marked as answer]
The phone has the latest updates. But I went back to the last stable release and its now working again perfectly. Something in the nightly since March didn't play nicely.

**Reply by jamorham** [reply]
Can you confirm the exact version you are now running as the current stable release is 3rd May?

**Comment by dicko72** [comment]
Running xDrip-plus-20230503-e50dfdf now. Was running nightly snapshot for
years prior to this.

**Comment by dicko72** [comment]
Actually looks like it may have started failing around 22nd May. Which
makes more sense as the latest stable works.

**Reply by jamorham** [reply]
If you get a chance could you see if the latest nightly resolves the problem? You should be able to easily just install any version over the top of another and change between them without problems.

**Comment by dicko72** [comment]
No worries, will try it and let you know.

**Comment by dicko72** [comment]
So far so good, will leave it running overnight and report back tomorrow.

**Reply by Navid200** [reply]
That's wonderful.
It will be great if you could log into GitHub and post in GitHub instead of responding to email notifications from GitHub.
For those who read these forums with less advanced devices, the thread looks really cluttered because of all the content your email client is including with every one of your replies.

**Comment by dicko72** [comment]
Appears to be fixed now in the latest nightly. Thank you so much for your help.
